verilog-to-routing/vtr-verilog-to-routing
# Add flat version of sync_netlists_to_routing

Run flat changes the internal cluster routing, and can also rotate truth tables in LUTs. To generate a bitstream or perform post-routing simulation, we need to update the intra-cluster routing data and mapping data of what inputs/outputs have been re-ordered by the router. this PR does that.

#### Description
Made a sync_netlists_to_routing flat routine which rebuilds the clustered netlist and the intra-cluster routing (pb_routes) and updates the atom lookup data structure.

#### Related Issue
Issue from OpenFPGA repo: 

#### Motivation and Context
Flat routing won't be usable by QuickLogic, OpenFPGA etc. without this change.

#### How Has This Been Tested?
- Ran with a QuickLogic architecture and verified the bitstream is correct
- Ran with the flagship architecture and confirmed it doesn't crash.
TODO:
- Run with stratixiv_arch, confirm it doesn't crash (currently it does)
- Run post-routing simulation on both flagship arch and stratixiv_arch to confirm they work.
- Should put at least one run-flat post-routing simulation for flagship arch and stratix iv arch in CI.

#### Types of changes

#### Checklist:
- No documentation changes
